src/Discord.Net.WebSocket/DiscordSocketClient.cs

```diff
@@ -1631,7 +1631,10 @@ internal SocketGuild RemoveGuild(ulong id)
 
         internal ISocketPrivateChannel AddPrivateChannel(API.Channel model, ClientState state)
         {
-            var channel = SocketChannel.CreatePrivate(this, state, model);
+            ISocketPrivateChannel channel;
+            if ((channel = state.GetChannel(model.Id) as ISocketPrivateChannel) != null)
```

Comment:
🤔 Literally doing a longer form of `if (state.GetChannel(model.Id) is ISocketPrivateChannel channel)`